scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -647,15 +647,41 @@ def constr_f2(x):
 
         # for multiple constraints the constraint violations should
         # be concatenated.
-        cv = solver._constraint_violation_fn([1.2, 1.])
-        assert_almost_equal(cv, [0.3, 0.64, 0])
+        xs = [(1.2, 1), (2.0, 2.0), (0.5, 0.5)]
+        vs = [(0.3, 0.64, 0.0), (2.1, 4.2, 0.0), (0, 0, 0)]
+
+        for x, v in zip(xs, vs):
+            cv = solver._constraint_violation_fn(np.array(x))
+            assert_almost_equal(cv, np.atleast_2d(v))
```

Comment:
Optional since it's the current state, but should use `assert_allclose` for new code.